lib/auth/Vault.ts

```diff
@@ -195,7 +205,8 @@ export default class Vault {
     authenticateV4Request(
         params: AuthV4RequestParams,
         requestContexts: any[] | null,
-        callback: (err: Error | null, data?: any) => void
+        callback: (err: Error | null, data?: any) => void,
+        options: AuthenticationOptions = {},
     ) {
```

Comment:
should keep the "standard" order of args, with callback coming last (and thankfully TS will help detect any inconsistency)

```suggestion
        options: AuthenticationOptions = {},
        callback: (err: Error | null, data?: any) => void,
    ) {
```